pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java

```diff
@@ -177,6 +177,12 @@
     public static final String CONFIG_OF_BROKER_GROUPBY_TRIM_THRESHOLD = "pinot.broker.groupby.trim.threshold";
     public static final int DEFAULT_BROKER_GROUPBY_TRIM_THRESHOLD = 1_000_000;
 
+    public static final String CONFIG_OF_BROKER_CLIENT_PROTOCOL = "pinot.broker.client.protocol";
```

Comment:
In order to support backward compat and migration, you should introduce config so that both http and https can be supported, both for client and netty. The way installations can migrate brokers by enabling both http and https, changing clients to https, ad then disabling https on the broker port.

For netty is a bit trickier. We need to upgrade the servers to support both, change the broker to start using https, and then disable the servers to use http.

Reply:
controllers, brokers, and servers now support multi-ingress for different protocol. this opens up a three-phase migration path by first enabling TLS-secured servers, then upgrading client connections to TLS, and finally shutting off unsecured server ingress.